### DISCUSSION TITLE
Unexpected internal error: [Errno 2] No such file or directory: 'npx'

### DISCUSSION THREAD

**Discussion by fangorntb** [airtai/fastkafka#3] [Q&A]
I really like this framework, but one mistake haunts me. Tell us how to fix it:
Unexpected internal error: [Errno 2] No such file or directory: 'npx'

**Comment by davorrunje** [comment]
Thank you for your interest in the framework.

Before trying to create docs, you should install dependencies for creating docs using the following CLI command:

```sh
fastkafka docs install_deps
``` 

However, if you already have `Node.js` installed but not `npx` not, you will get the following error:

```
Unexpected internal error: [Errno 2] No such file or directory: 'npx'
```

I am sorry for this bug. It is related to the way we install Node.js and its dependencies. We just managed to reproduce the bug you have locally and I created an issue https://github.com/airtai/fastkafka/issues/158.

We will fix it by the end of next week latest.

**Reply by fangorntb** [reply]
Also, how can I skip a commit if a critical error has occurred? I would like this message to return in the topic and be read, for example, after a restart. It would be cool if the problem of message loss was solved.

**Reply by davorrunje** [reply]
We are working on more robust error handling right now. Right now, we log all exceptions and move on. The new error handling is scheduled for release in four weeks if everything goes to plan.
